packages/fxa-auth-server/lib/routes/account.js

```diff
@@ -1173,70 +1173,44 @@ module.exports = (
-              password = new Password(authPW, authSalt, config.verifierVersion);
-              return password.verifyHash();
-            })
-            .then(verifyHashData => {
-              verifyHash = verifyHashData;
-
-              return setupKb();
-            })
-            .then(() => {
-              return db.resetAccount(accountResetToken, {
-                authSalt,
-                verifyHash,
-                wrapWrapKb,
-                verifierVersion: password.version,
-              });
-            })
-            .then(() => {
-              // Delete all passwordChangeTokens, passwordForgotTokens and
-              // accountResetTokens associated with this uid
-              return db.resetAccountTokens(accountResetToken.uid);
-            })
-            .then(() => {
-              // Notify the devices that the account has changed.
-              request.app.devices.then(devices =>
-                push.notifyPasswordReset(accountResetToken.uid, devices)
-              );
-
-              return db
-                .account(accountResetToken.uid)
-                .then(accountData => (account = accountData));
-            })
-            .then(() => {
-              return P.all([
-                request.emitMetricsEvent('account.reset', {
-                  uid: account.uid,
-                }),
-                log.notifyAttachedServices('reset', request, {
-                  uid: account.uid,
-                  generation: account.verifierSetAt,
-                }),
-                customs.reset(account.email),
-              ]);
-            });
-        }
-
-        function setupKb() {
+        async function resetAccountData() {
+          const authSalt = await random.hex(32);
+          password = new Password(authPW, authSalt, config.verifierVersion);
+          verifyHash = await password.verifyHash();
           if (recoveryKeyId) {
             // We have the previous kB, just re-wrap it with the new password.
-            return password.wrap(wrapKb).then(result => (wrapWrapKb = result));
+            wrapWrapKb = await password.wrap(wrapKb);
           } else {
             // We need to regenerate kB and wrap it with the new password.
-            return random.hex(32).then(result => {
-              wrapWrapKb = result;
-              return password
-                .unwrap(wrapWrapKb)
-                .then(result => (wrapKb = result));
-            });
+            wrapWrapKb = await random.hex(32);
+            wrapKb = await password.unwrap(wrapWrapKb);
           }
+          // db.resetAccount() deletes all the devices saved in the account,
+          // so grab the list to notify before we call it.
+          const devicesToNotify = await request.app.devices;
+          // Reset the account, and delete any other outstanding passwordChangeTokens,
+          // passwordForgotTokens or accountResetTokens associated with that account.
+          await db.resetAccount(accountResetToken, {
```

Comment:
I think the comment got outdated, might be better to remove or update to `account related tokens`.